packages/taiko-client/proposer/proposer.go

```diff
@@ -518,3 +610,118 @@ func (p *Proposer) sendTx(ctx context.Context, txCandidate *txmgr.TxCandidate) e
 func (p *Proposer) Name() string {
 	return "proposer"
 }
+
+// isProfitable checks if a transaction list is profitable to propose
+// Profitability is determined by comparing the revenue from transaction fees
+// to the costs of proposing and proving the block.
+func (p *Proposer) isProfitable(txLists []types.Transactions, proposingCosts *big.Int) (bool, error) {
+	totalTransactionFees, err := p.calculateTotalL2TransactionsFees(txLists)
+	if err != nil {
+		return false, err
+	}
+
+	costs, err := p.estimateTotalCosts(proposingCosts)
+	if err != nil {
+		return false, err
+	}
+
+	log.Debug("isProfitable", "total L2 fees", totalTransactionFees, "total L1 costs", costs)
+
+	return totalTransactionFees.Cmp(costs) > 0, nil
+}
+
+func (p *Proposer) calculateTotalL2TransactionsFees(txLists []types.Transactions) (*big.Int, error) {
+	totalGasConsumed := new(big.Int)
+
+	for _, txs := range txLists {
+		for _, tx := range txs {
+			baseFee := get75PercentOf(tx.GasFeeCap())
+			multiplier := new(big.Int).Add(tx.GasTipCap(), baseFee)
+			gasConsumed := new(big.Int).Mul(multiplier, baseFee)
+			totalGasConsumed.Add(totalGasConsumed, gasConsumed)
+		}
+	}
+
+	return totalGasConsumed, nil
+}
+
+func get75PercentOf(num *big.Int) *big.Int {
```

Comment:
The 75 percent of base fee sharing is the config in contract, let's use the chain config instead of a constant value.